src/main/java/it/gov/pagopa/pu/debtpositions/service/massive/DebtPositionMassiveServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+    this.installmentSynchronizeMapper = installmentSynchronizeMapper;
+    this.debtPositionRepository = debtPositionRepository;
+    this.debtPositionTypeOrgRepository = debtPositionTypeOrgRepository;
+    this.debtPositionTypeRepository = debtPositionTypeRepository;
+    this.organizationSearchClient = organizationSearchClient;
+    this.insertionActionMassiveDebtPositionService = insertionActionMassiveDebtPositionService;
+    this.updateActionMassiveDebtPositionService = updateActionMassiveDebtPositionService;
+  }
+
+
+  @Override
+  public String installmentSynchronize(InstallmentSynchronizeDTO installmentSynchronizeDTO, Boolean massive, String accessToken, String operatorExternalUserId) {
+    DebtPositionDTO debtPositionSynchronizeDTO = createDpFromInstallmentSynchronize(installmentSynchronizeDTO, accessToken);
+
+    DebtPosition debtPosition = checkProcessableDebtPosition(debtPositionSynchronizeDTO.getIupdOrg());
+
+    String workflowId = null;
+    InstallmentSynchronizeDTO.ActionEnum action = installmentSynchronizeDTO.getAction();
+    switch (action) {
+      case InstallmentSynchronizeDTO.ActionEnum.I ->
+        workflowId = insertionActionMassiveDebtPositionService.handleInsertion(debtPositionSynchronizeDTO, debtPosition, massive, accessToken, operatorExternalUserId);
+      case InstallmentSynchronizeDTO.ActionEnum.M ->
+        workflowId = updateActionMassiveDebtPositionService.handleUpdate(debtPositionSynchronizeDTO, massive,  accessToken);
+      case InstallmentSynchronizeDTO.ActionEnum.A ->
+        workflowId = updateActionMassiveDebtPositionService.handleCancellation(debtPositionSynchronizeDTO, massive, accessToken);
+    }
+
+    return workflowId;
+  }
+
+  private DebtPositionTypeOrg retrieveDebtPositionTypeOrg(Long organizationId, String debtPositionTypeCode) {
+    return debtPositionTypeOrgRepository.findByOrganizationIdAndCode(
+        organizationId, debtPositionTypeCode)
+      .orElseThrow(() -> new InvalidValueException(String.format("The debt position type code %s is not valid for this organizationId %s", debtPositionTypeCode, organizationId)));
+  }
+
+  private void populateFirstTransfer(InstallmentSynchronizeDTO installmentSynchronizeDTO, String accessToken, DebtPositionTypeOrg debtPositionTypeOrg) {
+    Long organizationId = installmentSynchronizeDTO.getOrganizationId();
+
+    Organization organization = organizationSearchClient.findByOrganizationId(organizationId, accessToken);
+    String category = debtPositionTypeRepository.findByDebtPositionTypeId(debtPositionTypeOrg.getDebtPositionTypeId()).getTaxonomyCode();
+
+    BigDecimal totalAmountOtherTransfers = installmentSynchronizeDTO.getTransfersList().stream()
+      .filter(Objects::nonNull).map(TransferSynchronizeDTO::getAmount)
+      .filter(Objects::nonNull).reduce(BigDecimal.ZERO, BigDecimal::add);
```

Comment:
is amount a BigDecimal? If yes, let's change it into Long and rename amount into amountCents
This reduce operation could be simplified in:
```suggestion.mapToLong(TransferSynchronizeDTO::getAmount)
      .sum();
```

ps: could getAmount really be null?